instrumentation/micrometer/micrometer-1.5/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/micrometer/v1_5/OpenTelemetryTimer.java

```diff
@@ -0,0 +1,167 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.micrometer.v1_5;
+
+import static io.opentelemetry.javaagent.instrumentation.micrometer.v1_5.Bridging.description;
+import static io.opentelemetry.javaagent.instrumentation.micrometer.v1_5.Bridging.toAttributes;
+
+import io.micrometer.core.instrument.AbstractTimer;
+import io.micrometer.core.instrument.Clock;
+import io.micrometer.core.instrument.Measurement;
+import io.micrometer.core.instrument.distribution.DistributionStatisticConfig;
+import io.micrometer.core.instrument.distribution.NoopHistogram;
+import io.micrometer.core.instrument.distribution.TimeWindowMax;
+import io.micrometer.core.instrument.distribution.pause.PauseDetector;
+import io.micrometer.core.instrument.util.TimeUtils;
+import io.opentelemetry.api.common.Attributes;
+import io.opentelemetry.api.metrics.LongHistogram;
+import io.opentelemetry.api.metrics.Meter;
+import java.util.Collections;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicLong;
+
+final class OpenTelemetryTimer extends AbstractTimer implements RemovableMeter {
+
+  // TODO: use bound instruments when they're available
+  private final LongHistogram otelHistogram;
+  private final Attributes attributes;
+  private final Measurements measurements;
+
+  volatile boolean removed = false;
+
+  OpenTelemetryTimer(
+      Id id,
+      Clock clock,
+      DistributionStatisticConfig distributionStatisticConfig,
+      PauseDetector pauseDetector,
+      Meter otelMeter) {
+    super(id, clock, distributionStatisticConfig, pauseDetector, TimeUnit.MILLISECONDS, false);
+
+    this.otelHistogram =
+        otelMeter
+            .histogramBuilder(id.getName())
+            .setDescription(description(id))
+            .setUnit("milliseconds")
+            .ofLongs()
+            .build();
+    this.attributes = toAttributes(id.getTags());
+
+    if (isUsingMicrometerHistograms()) {
+      measurements = new MicrometerHistogramMeasurements(clock, distributionStatisticConfig);
```

Comment:
Micrometer histograms are configurable via the API, the OTel ones aren't - in this PR I left the possibility to define "micrometer-style" histograms using micrometer settings (this is very similar to how `DropwizardMeterRegistry` works), completely separately from otel histograms.